### PR TITLE
fix(authService): redirect on refresh token failure when configured to do so

### PR DESCRIPTION
**Pull request by gregleon**
Proposed solution to #321 issue

### CODE REVIEW

**Comment by RWOverdijk** [comment]
@gregleon Awesome! :) Would you mind just changing the commit message according to our [contributing guidelines](https://github.com/SpoonX/about/blob/master/CONTRIBUTING.md#commit) and also change the coding style? I've commented the changes inline.

File: src/authService.js [hunk 1]
```diff
@@ -391,10 +391,22 @@ export class AuthService {
                                             : this.config.loginUrl), content, this.config.getOptionsForTokenRequests())
         .then(response => {
           this.setResponseObject(response);
-          this.authentication.resolveUpdateTokenCallstack(this.isAuthenticated());
+          if(this.getAccessToken()){
```
Comment (RWOverdijk) [inline src/authService.js]: if (this.getAccessToken()) {

File: src/authService.js [hunk 2]
```diff
@@ -391,10 +391,22 @@ export class AuthService {
...
+                this.authentication.resolveUpdateTokenCallstack(this.isAuthenticated());
```
Comment (RWOverdijk) [inline src/authService.js]: 2 spaces

File: src/authService.js [hunk 3]
```diff
@@ -391,10 +391,22 @@ export class AuthService {
...
+          }else{
```
Comment (RWOverdijk) [inline src/authService.js]: } else {

File: src/authService.js [hunk 4]
```diff
@@ -391,10 +391,22 @@ export class AuthService {
...
+              this.setResponseObject(null);
+              if (this.config.expiredRedirect) {
```
Comment (RWOverdijk) [inline src/authService.js]: Newline before if

File: src/authService.js [hunk 5]
```diff
@@ -391,10 +391,22 @@ export class AuthService {
...
+                PLATFORM.location.assign(this.config.expiredRedirect);
+              }
+              this.authentication.resolveUpdateTokenCallstack(Promise.reject(new Error('accessToken not found in refreshToken response')));
```
Comment (RWOverdijk) [inline src/authService.js]: Newline before

File: src/authService.js [hunk 6]
```diff
@@ -391,10 +391,22 @@ export class AuthService {
...
+          }
+
         })
         .catch(error => {
           this.setResponseObject(null);
+          if (this.config.expiredRedirect) {
```
Comment (RWOverdijk) [inline src/authService.js]: Newline before

File: src/authService.js [hunk 7]
```diff
@@ -391,10 +391,22 @@ export class AuthService {
...
+            PLATFORM.location.assign(this.config.expiredRedirect);
```
Comment (RWOverdijk) [inline src/authService.js]: Newline before

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

**Comment by RWOverdijk** [comment]
**Re: Commit message**

`fix(authService): redirect on refresh token failure when configured to do so`

**Review by gregleon — COMMENTED** [review]
I think I got all the changes covered in new commit

**Review by gregleon — COMMENTED** [review]
Done

**Review by RWOverdijk — APPROVED** [review]

**Comment by RWOverdijk** [comment]
Looks good to me. I'll get this in with the next release :)

**Comment by RWOverdijk** [comment]
Published!

**Pull request merged**